### PR TITLE
chore: convert CDK Pipelines to CodePipeline in CDK v2

### PR DESCRIPTION
**Pull request by philasmar**
*Issue #, if available:*
DOTNET-6296

*Description of changes:*
* Removed the need for an external role when doing cross account deployments.
* Converted CDKPipelines to CodePipelines in CDK v2

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: LambdaRuntimeDockerfiles/Infrastructure/README.md [hunk 1]
```diff
@@ -12,97 +12,34 @@ Infrastructure project allows to create pipeline to build and push .NET 5 Lambda
...
-For consistency purpose, same ECR repository name is used across all ECRs.
+#### Notes
+ - AWS Profiles used to execute `bootstrap.ps1` must have administrator access.
+ - All resources used to bootstrap the pipeline must already exist.
+ - `AccountId` is AWS AccountId used for deploying CDK App.
+ - `CodeCommitAccountId` is AWS AccountId that contains source repository.
+ - If the CodeCommit repository is in the same account, use the same account Id for `PipelineAccountId` and `CodeCommitAccountId`.
+ - When doing a cross-account deployment, you need to have AWS Profiles for both accounts.
+ - `bootstrap.ps1` will run 2 `cdk bootstrap` commands for the cross account deployments to establish a trust relationships between the accounts. This way, we do not require a separate IAM role to be created manually.
+ - For consistency purpose, same ECR repository name is used across all ECRs.
```
Comment (normj) [inline LambdaRuntimeDockerfiles/Infrastructure/README.md:42]: This note doesn't make a lot of sense. I assume it means the same repository name is used no matter what AWS account is being used but it is worded weird.

File: LambdaRuntimeDockerfiles/Infrastructure/src/Infrastructure/Infrastructure.csproj [hunk 2]
```diff
@@ -9,11 +9,8 @@
 
   <ItemGroup>
     <!-- CDK Construct Library dependencies -->
-    <PackageReference Include="Amazon.CDK" Version="1.74.0" />
-    <PackageReference Include="Amazon.CDK.Pipelines" Version="1.74.0" />
-    <PackageReference Include="Amazon.CDK.AWS.CodePipeline" Version="1.74.0" />
-    <PackageReference Include="Amazon.CDK.AWS.CodeCommit" Version="1.74.0" />
-    <PackageReference Include="Amazon.CDK.AWS.CodePipeline.Actions" Version="1.74.0" />
+	<PackageReference Include="Amazon.CDK.Lib" Version="2.44.0" />
+	<PackageReference Include="Constructs" Version="[10.0.0,11.0.0)" />
```
Comment (normj) [inline LambdaRuntimeDockerfiles/Infrastructure/src/Infrastructure/Infrastructure.csproj:13]: nit: Do we have to have a dependency on `Constructs`? Seems like that is brought in via `Amazon.CDK.Lib`.

**Review by normj — APPROVED** [review]
A couple small nit but I'm good with the change

File: LambdaRuntimeDockerfiles/Infrastructure/README.md (hunk 1 above)
Comment (philasmar) [reply LambdaRuntimeDockerfiles/Infrastructure/README.md:42]: Are you talking about the whole block? or the last bullet point?

Comment (normj) [reply LambdaRuntimeDockerfiles/Infrastructure/README.md:42]: Just the last bullet point

**Review by 96malhar — APPROVED** [review]

File: LambdaRuntimeDockerfiles/Infrastructure/src/Infrastructure/Infrastructure.csproj (hunk 2 above)
Comment (philasmar) [reply LambdaRuntimeDockerfiles/Infrastructure/src/Infrastructure/Infrastructure.csproj]: Removed

**Review by ashovlin — APPROVED** [review]

**Pull request merged**